packages/yoastseo/src/morphology/dutch/stem.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * @file Dutch stemming algorithm. Adapted from:
+ * @author:
+ * @copyright
+ * All rights reserved.
+ * Implementation of the stemming algorithm from http://snowball.tartarus.org/algorithms/dutch/stemmer.html
+ * Copyright of the algorithm is: Copyright (c) 2001, Dr Martin Porter and can be found at http://snowball.tartarus.org/license.php
+ *
+ * Redistribution and use in source and binary forms, with or without modification, is covered by the standard BSD license.
+ */
+/**
+ *
+ * Determines the start index of the R1 region.
+ * R1 is the region after the first non-vowel following a vowel. It should include at least 3 letters.
+ *
+ * @param {string} word The word for which to determine the R1 region.
+ * @returns {number} The start index of the R1 region.
+ */
+const determineR1 = function( word ) {
+	// Start with matching the first cluster that consists of a vowel and a non-vowel.
+	let r1Index = word.search( /[aeiouyèäüëïöáéíóú][^aeiouyèäüëïöáéíóú]/ );
+	// Then add 2 since the R1 index is the index after the first vowel & non-vowel matched with the regex.
+	if ( r1Index !== -1 ) {
+		r1Index += 2;
+	}
+
+	// Adjust R1 so that the region preceding it includes at least 3 letters.
+	if ( r1Index !== -1 && r1Index < 3 ) {
+		r1Index = 3;
+	}
+
+	return r1Index;
+};
+
+
+/**
+ * Searches for suffixes in a word.
+ *
+ * @param {string} word 	The word in which to look for suffixes.
+ * @param {string} suffixStep	 One of the three steps of findings suffixes.
+ * @param {number} r1Index	 The index of the R1 region.
+ * @returns {object} The index of the suffix and extra information about whether, and how, the stem will need to be modified.
+ */
+const findSuffix = function( word, suffixStep, r1Index ) {
+	for ( const suffixClass in suffixStep ) {
+		if ( suffixStep.hasOwnProperty( suffixClass ) ) {
+			const suffixes = suffixStep[ suffixClass ].suffixes;
+
+			const matchedRegex = suffixes.find( suffixRegex => new RegExp( suffixRegex ).exec( word ) );
+
+			if ( matchedRegex ) {
+				const matched = new RegExp( matchedRegex ).exec( word );
+				const suffix = matched[ matched.length - 1 ];
+				const suffixIndex = word.lastIndexOf( suffix );
+
+				if ( suffixIndex >= r1Index ) {
+					return {
+						suffixIndex: suffixIndex,
+						stemModification: ( suffixStep[ suffixClass ].stemModification ),
```

Comment:
```suggestion
						stemModification: suffixStep[ suffixClass ].stemModification,
```
The brackets can be removed.